### PR TITLE
refactor: Applied Clippy suggestions

### PR DESCRIPTION
**Pull request by frol**
It is mostly a result of:

```
cargo fix --workspace --all-features --allow-dirty --edition-idioms
```

In the absolute numbers, most of the changes are around explicit usage of an unnamed lifetime. I have no preference here. Should we use it or not?

The rest are unnecessary clones and unnecessary `extern crate`.

### CODE REVIEW

**Comment by frol** [comment]
Well, Clippy is the defacto standard linter. And those hidden lifetimes are essentially a warning:

> warning: hidden lifetime parameters in types are deprecated

File: core/store/src/trie/nibble_slice.rs [hunk 1]
```diff
@@ -83,7 +83,7 @@ impl<'a> NibbleSlice<'a> {
     }
 
     /// Create a new nibble slice from the given HPE encoded data (e.g. output of `encoded()`).
-    pub fn from_encoded(data: &'a [u8]) -> (NibbleSlice, bool) {
+    pub fn from_encoded(data: &'a [u8]) -> (NibbleSlice<'_>, bool) {
```
Comment (bowenwang1996) [inline core/store/src/trie/nibble_slice.rs:86]: Is the lifetime parameter `'a` necessary here? Seems to me that it can be omitted.

Comment (frol) [reply core/store/src/trie/nibble_slice.rs:86]: @bowenwang1996 It is required:

```
error[E0621]: explicit lifetime required in the type of `data`
  --> core/store/src/trie/nibble_slice.rs:87:10
   |
86 |     pub fn from_encoded(data: &[u8]) -> (NibbleSlice<'a>, bool) {
   |                               ----- help: add explicit lifetime `'a` to the type of `data`: `&'a [u8]`
87 |         (Self::new_offset(data, if data[0] & 16 == 16 { 1 } else { 2 }), data[0] & 32 == 32)
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime `'a` required
```

Comment (bowenwang1996) [reply core/store/src/trie/nibble_slice.rs:86]: Oh nvm I didn't realize that this is under `impl<'a> NibbleSlice<'a>`.

**Pull request merged**